display_list/display_list_color_source.h

```diff
@@ -221,39 +221,47 @@ class DlImageColorSource final : public SkRefCnt,
 
   std::shared_ptr<DlColorSource> with_sampling(
       DlImageSampling sampling) const override {
-    return std::make_shared<DlImageColorSource>(
-        sk_image_, horizontal_tile_mode_, vertical_tile_mode_, sampling,
-        matrix_ptr());
+    return std::make_shared<DlImageColorSource>(image_, horizontal_tile_mode_,
+                                                vertical_tile_mode_, sampling,
+                                                matrix_ptr());
   }
 
   DlColorSourceType type() const override { return DlColorSourceType::kImage; }
   size_t size() const override { return sizeof(*this); }
 
-  bool is_opaque() const override { return sk_image_->isOpaque(); }
+  bool is_opaque() const override {
+    if (!image_->skia_image()) {
+      return false;
+    }
+    return image_->skia_image()->isOpaque();
```

Comment:
nit: consider implementing `isOpaque` on `DlImage`. It will be necessary to support the `DlDeferredGpuImage` down the road.

Reply:
But also feel free to land this as-is and I'll fix that up when I come back through here :)

Reply:
Issue filed https://github.com/flutter/flutter/issues/109286 :)